parquet/src/file/writer.rs

```diff
@@ -171,7 +183,30 @@ impl<W: Write + Send> SerializedFileWriter<W> {
     /// Creates new file writer.
     pub fn new(buf: W, schema: TypePtr, properties: WriterPropertiesPtr) -> Result<Self> {
         let mut buf = TrackedWrite::new(buf);
-        Self::start_file(&mut buf)?;
+
+        #[cfg(feature = "encryption")]
+        let file_encryptor = match properties.file_encryption_properties.as_ref() {
+            None => None,
+            Some(encryption_props) => Some(Arc::new(FileEncryptor::new(encryption_props.clone())?)),
+        };
+
+        #[cfg(feature = "encryption")]
+        if properties.file_encryption_properties.is_some() {
+            properties
+                .file_encryption_properties
+                .clone()
+                .unwrap()
+                .validate_encrypted_column_names(SchemaDescriptor::new(schema.clone()))?;
+        }
```

Comment:
```suggestion
        let schema_descriptor = SchemaDescriptor::new(schema.clone());

        #[cfg(feature = "encryption")]
        if let Some(file_encryption_properties) = properties.file_encryption_properties.as_ref() {
            file_encryption_properties.validate_encrypted_column_names(&schema_descriptor)?;
        }
```

Plus change below to:
```
Ok(Self {
            buf,
            schema: schema,
            descr: schema_descriptor,
```

Reply:
Do you mean:
```rust
        Ok(Self {
            buf,
            schema: schema.clone(),
            descr: Arc::new(schema_descriptor),
```

Reply:
Yes that bit should be able to be changed if you create the schema_descriptor above here.

Reply:
Oh I see you added the `Arc::new(`. Yeah that's needed, although the `schema.clone()` should be able to just be `schema`.

Reply:
Good point about the schema. Changed.